File: math/calculation/possibilityFinder.c
#include<stdio.h>

int factorial(int i){
	for(int j=i-1;j>0;j--){
		i *= j;
	}
return i;
}

int main(){
	int total,pieces,possibilities;
	printf("Number of Possibilities\nHow many objects are there: ");
	scanf("%d",&pieces);
	printf("How many total spaces are there including ones with pieces on them: ");
	scanf("%d",&total);

	possibilities = factorial(total)/(factorial(total-3)*factorial(pieces));
	printf("There are %d unique possibilities\n",possibilities);
}

File: scripts/python/possibilityFinder.py
def factorial(n):
	total=1
	for x in range(n):
		total *= (n-x)
	return total

total=pieces=possibilities=0
pieces = input("Number of Possibilities\nHow many objects are there: ")

total = input("How many total spaces are there including ones with pieces on them: ")

possibilities = factorial(total)/(factorial(total-3)*factorial(pieces));
print "There are %d unique possibilities"%possibilities
